electroncash/rpa/paycode.py

```diff
@@ -243,15 +243,16 @@ def generate_transaction_from_paycode(wallet, config, amount, rpa_paycode=None,
         # Generate the raw transaction
         raw_tx_string = tx.as_dict()["hex"]
 
-        # Get the TxId for this raw Tx.
-        double_hash_tx = bytearray(sha256(sha256(bytes.fromhex(raw_tx_string))))
-        double_hash_tx.reverse()
-        txid = double_hash_tx.hex()
+        # Get the 'txinid' (double hash of serialized input) for this raw Tx.
+        txin= tx._inputs[0]
+        txin_script = Transaction.input_script(txin,False,sign_schnorr = True)
+        serialized_signed_input = Transaction.serialize_input(txin,txin_script)
+        double_hash_txin = bytearray(sha256(sha256(bytes.fromhex(serialized_signed_input))))
+        txinid = double_hash_tx.hex()
```

Comment:
You could probably make this slow grinding routine 2x faster by not working with hex chars but with bytes directly .. but then you would have to compare half-bytes using bit-ops....

Reply:
The whole grinding operation needs optimization beyond this.